code/game/objects/items/motion_detector.dm

```diff
@@ -331,3 +331,104 @@
 		icon_state = "minidetector_on_[detector_mode]"
 	else
 		icon_state = "minidetector_off"
+
+/obj/item/motiondetector/integrated
+	name = "Integrated tactical sensor"
+	desc = "A device that detects hostile movement; this one is specially integrated for M3-UL pattern armor. Hostiles appear as red blips. Friendlies with the correct IFF signature appear as green, and their bodies as blue, unrevivable bodies as dark blue. It has a mode selection interface."
+	ping = FALSE //Stealth modo
+
+/obj/item/motiondetector/integrated/process()
```

Comment:
Is this proc copypasted? You should avoid copypasting long bits of code, use overrides instead especially since this is a child of the main motion detector

Reply:
Yeah sadly. I'm currently a noob when it comes to coding original things, but atleast it gives the bare essentials.

Reply:
By default it will inherit the parent /process proc, meaning that you don't have redefine it entirely. BYOND is object oriented and child objects inherit everything from their parent objects, in this case the normal motion detector.

Reply:
It does have its own tab, however there's one problem. If the armor is disequipped and reequipped, the motion detector effect will be gone, even with the armor on (Line 188 - 184). On some casesif the armor is disequipped, the motion detector effect will still happen regardless (Line 175 - Line 178).

The new `/obj/item/motiondetector/process()` is meant to remove the `if(get_turf(src) != get_turf(operator))` line, which is used if the operator of the motion detector is kept inside in a container, which the motion detector will deactivate in the process. I tried a rudimentary way of disabling the line by creating a new `/obj/item/motiondetector/integrated/process()` since it's a child of the original motion detector, I don't know what else to do about it.

Reply:
Remove this proc